packages/amplication-client/src/VersionControl/Build.tsx

```diff
@@ -238,6 +234,48 @@ const Build = ({ build, onError, open }: Props) => {
 
 export default Build;
 
+type BuildHeaderProps = {
+  build: models.Build;
+  deployments: models.Deployment[] | null;
+};
+
+const BuildHeader = ({ build, deployments }: BuildHeaderProps) => {
+  const account = build.createdBy?.account;
+
+  const deployedClassName = `${CLASS_NAME}__header--deployed`;
+
+  return (
+    <div
+      className={`${CLASS_NAME}__header ${deployments && deployedClassName} `}
+    >
+      <h3>
+        Version<span>{build.version}</span>
+      </h3>
+      {deployments ? (
+        <>
+          <Icon icon="publish" />
+          <a href={deployments[0].environment.address}>
+            <Icon icon="link_2" />
+          </a>
+        </>
+      ) : (
+        <>
+          <CircleIcon
+            size={EnumCircleIconSize.Small}
+            {...BUILD_STATUS_TO_STYLE[
+              build.status || models.EnumBuildStatus.Invalid
+            ]}
+          />
+
+          <span>{build.status}</span>
+        </>
+      )}
+      <span className="spacer" />
+      <UserAndTime account={account} time={build.createdAt} />
+    </div>
+  );
+};
```

Comment:
Can you extract this to another file?

Reply:
This component will never be used without the build component and it is even not exported. Do we really want to have it in a separate file?

Reply:
Yes, any component more than a few lines long better exist in a component inside the directory of the main component.

Reply:
This is not a show stopper. Please handle whenever you have the time.